File: content/problems/0011-container-with-most-water/solutions/two-pointers/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = (int)height.size() - 1, best = 0;
        while (left < right) {
            best = max(best, (right - left) * min(height[left], height[right]));
            if (height[left] <= height[right]) ++left; else --right;
        }
        return best;
    }
};

File: content/problems/0011-container-with-most-water/solutions/two-pointers/Solution.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1, best = 0;
        while (left < right) {
            best = Math.max(best, (right - left) * Math.min(height[left], height[right]));
            if (height[left] <= height[right]) left++; else right--;
        }
        return best;
    }
}
